cirq/ops/parallel_gate_operation_test.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import numpy as np
+import cirq
+
+
+def test_parallel_gate_operation_init():
+    q = cirq.NamedQubit('q')
+    r = cirq.NamedQubit('r')
+    g = cirq.SingleQubitGate()
+    v = cirq.ParallelGateOperation(g, (q, r))
+    assert v.gate == g
+    assert v.qubits == (q, r)
+
+
+def test_invalid_parallel_gate_operation():
+    three_qubit_gate = cirq.ThreeQubitGate()
+    single_qubit_gate = cirq.SingleQubitGate()
+    repeated_qubits = [cirq.GridQubit(0, 0),cirq.GridQubit(0, 0)]
+    with pytest.raises(ValueError):
+        cirq.ParallelGateOperation(three_qubit_gate, cirq.NamedQubit("a"))
+    with pytest.raises(ValueError):
+        cirq.ParallelGateOperation(single_qubit_gate, repeated_qubits)
+
+
+def test_gate_operation_eq():
+    g1 = cirq.SingleQubitGate()
+    g2 = cirq.SingleQubitGate()
+    r1 = [cirq.NamedQubit('r1')]
+    r2 = [cirq.NamedQubit('r2')]
+    r12 = r1 + r2
+    r21 = r2 + r1
+
+    eq = cirq.testing.EqualsTester()
+    eq.make_equality_group(lambda: cirq.ParallelGateOperation(g1, r1))
+    eq.make_equality_group(lambda: cirq.ParallelGateOperation(g2, r1))
+    eq.add_equality_group(cirq.ParallelGateOperation(g1, r12),
+                          cirq.ParallelGateOperation(g1, r21))
+
+
+def test_with_qubits_and_transform_qubits():
+    g = cirq.SingleQubitGate()
+    op = cirq.ParallelGateOperation(g, cirq.LineQubit.range(3))
+    assert op.with_qubits(*cirq.LineQubit.range(3, 0, -1)) \
+        == cirq.ParallelGateOperation(g, cirq.LineQubit.range(3, 0, -1))
+    assert op.transform_qubits(lambda e: cirq.LineQubit(-e.x)
+                               ) == cirq.ParallelGateOperation(g,
+                                                           [cirq.LineQubit(0),
+                                                           cirq.LineQubit(-1),
+                                                           cirq.LineQubit(-2)])
+
+
+def test_decompose():
+    qreg = cirq.LineQubit.range(3)
+    op = cirq.ParallelGateOperation(cirq.X, qreg)
+    assert cirq.decompose(op) == cirq.X.on_each(qreg)
```

Comment:
Test failure because we switched up the signature of on_each

Reply:
Fixed